src/test/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPJwtKeyByOpenIdConnectAuthenticatorTest.java

```diff
@@ -58,6 +59,86 @@ public void basicTest() {
 		Assert.assertEquals(3, creds.getAttributes().size());
 	}
 
+
+	@Test
+	public void jwksUriTest() {
+		String requiredIssuer = "requiredIssuer";
+		String requiredAudience = "requiredAudience";
+		Settings settings = Settings.builder()
+			.put("jwks_uri", mockIdpServer.getJwksUri())
+			.put("required_issuer", requiredIssuer)
+			.put("required_audience", requiredAudience)
+			.build();
+
+		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
+
+		AuthCredentials creds = jwtAuth.extractCredentials(new FakeRestRequest(
+			ImmutableMap.of("Authorization", TestJwts.MC_COY_SIGNED_OCT_1), new HashMap<>()), null);
+
+		Assert.assertNotNull(creds);
+		Assert.assertEquals(TestJwts.MCCOY_SUBJECT, creds.getUsername());
+		Assert.assertEquals(TestJwts.TEST_AUDIENCE, creds.getAttributes().get("attr.jwt.aud"));
+		Assert.assertEquals(0, creds.getBackendRoles().size());
+		Assert.assertEquals(3, creds.getAttributes().size());
+		Assert.assertEquals(requiredAudience, jwtAuth.getRequiredAudience());
+		Assert.assertEquals(requiredIssuer, jwtAuth.getRequiredIssuer());
+	}
+
+	@Test
+	public void jwksMatchingRequiredIssuerInClaimTest() {
+		String requiredIssuer = "requiredIssuer";
+		Settings settings = Settings.builder()
+			.put("jwks_uri", mockIdpServer.getJwksUri())
+			.put("required_issuer", requiredIssuer)
+			.build();
+
+		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
+
+		AuthCredentials creds = jwtAuth.extractCredentials(new FakeRestRequest(
+			ImmutableMap.of("Authorization", TestJwts.MC_COY_SIGNED_OCT_2), new HashMap<>()), null);
+
+		Assert.assertNotNull(creds);
+		Assert.assertEquals(TestJwts.MCCOY_SUBJECT, creds.getUsername());
+		Assert.assertEquals(TestJwts.TEST_AUDIENCE, creds.getAttributes().get("attr.jwt.aud"));
+		Assert.assertEquals(0, creds.getBackendRoles().size());
+		Assert.assertEquals(4, creds.getAttributes().size());
+		Assert.assertFalse(creds.getAttributes().get("attr.jwt.iss").contains(jwtAuth.getRequiredIssuer()));
+	}
+
+	@Test
```

Comment:
I'm trying to understand, based on the equivalent test in [HTTPJwtAuthenticatorTest](https://github.com/opensearch-project/security/blob/main/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java#L356-L364) it looks like this should be returning `null` creds because if the audience claim in the token is different from the configured `requiredAudience` then it would not be considered a valid token. Since the creds are not null here, does that mean it would successfully authenticate?

Reply:
The reason that the one You found is providing null instead of `AuthCredentials` is that we are using different `extractCredentials0() `method (from `HTTPJwtAuthenticator` instead of `AbstractHTTPJwtAuthenticator`). The one more verbose is trying to parse claims and checks ie. audience, while the other one in the same time just takes claims from token without any checks there (or at least the checks agains audience/issuer):

`jwtParser.parseClaimsJws(jwtToken)` throws Exception which leads to `return null;` in line 205:
![image](https://github.com/opensearch-project/security/assets/18590055/c6276e99-741e-4d7e-a24f-c016fb9483be)

in the meantime `jwt.getClaims();` is just returning the claims which are just validated only against signature, expiry and nbf.

Reply:
The topic of this issue was to just provide jwks_uri to use it instead of changing signing key manually I guess, not to change current plugin behavior. Should we change this behavior within this issue?

Reply:
@cwperks I have to apologize for the irritation here. Of course, having `required_audience` and `required_issuer` as config attributes needs the accompanying logic. We will add this logic ASAP.

Reply:
Sure, now it's doable when I know what to do with those params 😉